### PR TITLE
fix(file-too-large): #CO-1748 send http error 413

### PR DESCRIPTION
**Pull request by Ozen40**
## Describe your changes
If you try to escalate a ticket that has a file too large for Jira, it will fail the escalation and send an http 413 error (Payload Too Large).
## Checklist tests

- Escalate a ticket with a file bigger than 10mb and check if it display the error and fail the escalation

## Issue ticket number and link
[CO-1748](https://jira.support-ent.fr/browse/CO-1748)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java [hunk 1]
```diff
@@ -155,18 +155,23 @@ private void doTicketEscalation(final JsonObject ticket, final JsonArray comment
...
+                                    .put(ATTACHMENT_CONTENT_FIELD, encodedData);
+                            attachments.add(attachment);
+
+                            // Create issue only if all attachments have been retrieved successfully
+                            if (successfulDocs.incrementAndGet() == attachmentsIds.size()) {
+                                EscalationServicePivotImpl.this.getDataAndCreateIssue( ticket, attachments,
+                                        finalComments, issue, escalationUser, handler);
+                            }
+                        } else {
+                            handler.handle(new Either.Left<>(Ticket.PAYLOADTOOLARGE));
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java:172]: Rajoute un log d'erreur aussi avec le résultat du metadata

File: src/main/java/net/atos/entng/support/constants/Ticket.java [hunk 2]
```diff
@@ -51,4 +51,8 @@ public class Ticket {
     public static final String STRUCTURE_ID = "structureId";
     public static final String USERID = "userId";
     public static final String AUTO_OPEN_TICKET = "support.ticket.auto.open";
+
+    public static final String METADATE = "metadata";
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/constants/Ticket.java:55]: erreur typo

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java [hunk 3]
```diff
@@ -155,18 +155,25 @@ private void doTicketEscalation(final JsonObject ticket, final JsonArray comment
...
-
-                        JsonObject attachment = new JsonObject().put(ATTACHMENT_NAME_FIELD, filename)
-                                .put(ATTACHMENT_CONTENT_FIELD, encodedData);
-                        attachments.add(attachment);
-
-                        // Create issue only if all attachments have been retrieved successfully
-                        if (successfulDocs.incrementAndGet() == attachmentsIds.size()) {
-                            EscalationServicePivotImpl.this.getDataAndCreateIssue( ticket, attachments,
-                                    finalComments, issue, escalationUser, handler);
+                        if (file.getDocument().getJsonObject(Ticket.METADATA).getInteger(Ticket.SIZE) < 10000000) {
```
Comment (qperie) [inline src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java:158]: Le 10000000 tu peux en faire un constante

**Review by qperie — APPROVED** [review]

File: src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java [hunk 4]
```diff
@@ -155,18 +157,25 @@ private void doTicketEscalation(final JsonObject ticket, final JsonArray comment
...
+                            attachments.add(attachment);
+
+                            // Create issue only if all attachments have been retrieved successfully
+                            if (successfulDocs.incrementAndGet() == attachmentsIds.size()) {
+                                EscalationServicePivotImpl.this.getDataAndCreateIssue( ticket, attachments,
+                                        finalComments, issue, escalationUser, handler);
+                            }
+                        } else {
+                            log.error(String.format("[Support@%s::doTicketEscalation] %s",
+                                    this.getClass().getSimpleName(), "Payload too large, metadata size : " + file.getDocument().getJsonObject(Ticket.METADATA).getInteger(Ticket.SIZE)));
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/services/impl/EscalationServicePivotImpl.java:175]: Sécurise ton code

```suggestion
 this.getClass().getSimpleName(), "Payload too large, metadata size : " + file.getDocument().getJsonObject(Ticket.METADATA, new JsonObject()).getInteger(Ticket.SIZE, 0)));
```

**Pull request merged**